### DISCUSSION TITLE
Maximum allowed torque for the last four actuators of the PSM

### DISCUSSION THREAD

**Discussion by yafei-ou** [jhu-dvrk/sawIntuitiveResearchKit#199] [Q&A]
Hi all,

I am wondering what the maximum allowed torque is for the last four actuators of the PSM (Classic da Vinci). I noticed this computation in [compute-ftmax.py](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/blob/356bac3e86a10468e05157ae996b86370eb29248/share/kinematic/compute-ftmax.py). This final result (0.3436) is in Newton meter, is that correct? Is this the actual (safe) amount of torque that the 4 actuators of the PSM can reach in practice? Could the robot be damaged if the actuators are kept working at close to this value for a long time (e.g., 1 hour)?

Thanks a lot for any help in advance.

**Comment by adeguet1** [comment]
Regarding the values you found, I can't recall why I wrote this script.  It was likely when we converted the PID controller to run on actuators instead of joints.  In the case of the PSM, there is no coupling matrix in the script so what you get is the max actuator torques, i.e. torques at the sterile adapter.  These are not the same as the joint maximum torques.  Each instrument has its own coupling matrix (see files in share/tool/*.json).  The cap on the maximum effort sent to motors is performed at the lowest level, i.e. in sawRobotIO1394.  See https://github.com/jhu-saw/sawRobotIO1394/blob/d5e652d624f515ccfe534d3aa4a0141a5c6c40eb/components/code/mtsRobot1394.cpp#L1577  The max values are pulled from the sawRobotIO XML configurations files and are provided in amperes.  These were provided by Intuitive in the dVRK User Guide.

Regarding a sustained use at high torques, I don't really know if that could hurt either the instrument or the arm itself.  Are you considering fast changes or something like holding an object for a long period?

**Reply by yafei-ou** [reply]
Hi Anton,

Thank you for the information.

> In the case of the PSM, there is no coupling matrix in the script so what you get is the max actuator torques, i.e. torques at the sterile adapter.

Yes, actually this is what I am looking for. In that case, the value I get (0.3436) is the max value that the actuators can generate, is that correct?
